validator/lib/validateI3dm.js

```diff
@@ -183,22 +188,89 @@ function validateI3dm(content) {
     }
 
-    if (defined(!featureTableJson.NORMAL_UP) && defined(featureTableJson.NORMAL_RIGHT)) {
+    if (!defined(featureTableJson.NORMAL_UP) && defined(featureTableJson.NORMAL_RIGHT)) {
         return 'Feature table property NORMAL_UP is required when NORMAL_RIGHT is present.';
     }
 
     if (defined(featureTableJson.NORMAL_UP_OCT32P) && !defined(featureTableJson.NORMAL_RIGHT_OCT32P)) {
         return 'Feature table property NORMAL_RIGHT_OCT32P is required when NORMAL_UP_OCT32P is present.';
     }
 
-    if (defined(!featureTableJson.NORMAL_UP_OCT32P) && defined(featureTableJson.NORMAL_RIGHT_OCT32P)) {
+    if (!defined(featureTableJson.NORMAL_UP_OCT32P) && defined(featureTableJson.NORMAL_RIGHT_OCT32P)) {
         return 'Feature table property NORMAL_UP_OCT32P is required when NORMAL_RIGHT_OCT32P is present.';
     }
 
     if (defined(featureTableJson.POSITION_QUANTIZED) && (!defined(featureTableJson.QUANTIZED_VOLUME_OFFSET) || !defined(featureTableJson.QUANTIZED_VOLUME_SCALE))) {
         return 'Feature table properties QUANTIZED_VOLUME_OFFSET and QUANTIZED_VOLUME_SCALE are required when POSITION_QUANTIZED is present.';
     }
 
+    var featureTable = new Cesium3DTileFeatureTable(featureTableJson, featureTableBinary);
+    var normalUp, normalRight;
+    var octUp = false;
+    var octRight = false;
+    var componentDatatype;
+    var i;
+
+    if (defined(featureTableJson.NORMAL_UP)) {
+        featureTable.featuresLength = featuresLength * 3;
+        componentDatatype = ComponentDatatype.fromName(defaultValue(featureTableJson.NORMAL_UP.componentType, 'FLOAT', 3));
+        normalUp = featureTable.getPropertyArray('NORMAL_UP', componentDatatype, 1);
+    } else if (defined(featureTableJson.NORMAL_UP_OCT32P)) {
+        octUp = true;
+        featureTable.featuresLength = featuresLength * 2;
+        componentDatatype = ComponentDatatype.fromName(defaultValue(featureTableJson.NORMAL_UP_OCT32P.componentType, 'UNSIGNED_SHORT', 2));
+        normalUp = featureTable.getPropertyArray('NORMAL_UP_OCT32P', componentDatatype, 1);
+    }
+
+    if (defined(featureTableJson.NORMAL_RIGHT)) {
+        featureTable.featuresLength = featuresLength * 3;
+        componentDatatype = ComponentDatatype.fromName(defaultValue(featureTableJson.NORMAL_RIGHT.componentType, 'FLOAT', 3));
+        normalRight = featureTable.getPropertyArray('NORMAL_RIGHT', componentDatatype, 1);
+    } else if (defined(featureTableJson.NORMAL_RIGHT_OCT32P)) {
+        octRight = true;
+        featureTable.featuresLength = featuresLength * 2;
+        componentDatatype = ComponentDatatype.fromName(defaultValue(featureTableJson.NORMAL_RIGHT_OCT32P.componentType, 'UNSIGNED_SHORT', 2));
+        normalRight = featureTable.getPropertyArray('NORMAL_RIGHT_OCT32P', componentDatatype, 1);
+    }
+
+    if(defined(normalUp) && defined(normalRight)) {
+        for(i = 0; i < featuresLength; i++) {
+            var normalUpVec3 = new Cartesian3();
```

Comment:
Put this outside of the for loop so that we only have to allocate a `Cartesian3` once.

Also rename to `normalUp`.